frontend/pages/collections/[slug].page.tsx

```diff
@@ -78,3 +103,71 @@ export const getStaticPaths: GetStaticPaths = async () => {
     fallback: "blocking",
   };
 };
+
+const MobileSharingButtons: FC<CollectionProps> = ({ collection }: CollectionProps) => {
+  const [showIcons, setShowIcons] = useState(false);
+  const onClick = () => {
+    setShowIcons((prevState) => !prevState);
+  };
+  return (
+    <Grid container justifyContent="flex-end">
+      <Button variant="contained" color="primary" onClick={onClick} sx={{ borderRadius: "50%" }}>
+        <ShareIcon />
+      </Button>
+      <>
+        {showIcons ? (
+          <Grid container justifyContent="flex-end">
+            <SharingButtonGroup collection={collection} />
+          </Grid>
+        ) : null}
+      </>
+    </Grid>
+  );
+};
+
+const DesktopSharingButtons: FC<CollectionProps> = ({ collection }: CollectionProps) => {
+  return (
+    <Grid container justifyContent="flex-end">
+      <SharingButtonGroup collection={collection} />
+    </Grid>
+  );
+};
+
+const SharingButtonGroup: FC<CollectionProps> = ({ collection }: CollectionProps) => {
+  return (
+    <>
+      <Grid
+        container
+        alignItems="center"
+        justifyContent="center"
+        spacing={2}
+        sx={{ m: 2, maxWidth: 200 }}
+      >
+        <Grid item>
+          <FacebookShareButton
+            url={`https://modularhistory.com/collections/${collection.slug}`}
+            quote={collection.title}
+          >
+            <FacebookIcon size={36} round={true} />
+          </FacebookShareButton>
+        </Grid>
+        <Grid item>
+          <TwitterShareButton
+            url={`https://modularhistory.com/collections/${collection.slug}`}
+            title={collection.title}
+          >
+            <TwitterIcon size={36} round={true} />
+          </TwitterShareButton>
+        </Grid>
+        <Grid item>
+          <EmailShareButton
+            url={`https://modularhistory.com/collections/${collection.slug}`}
+            subject={collection.title}
+          >
+            <EmailIcon size={36} round={true} />
+          </EmailShareButton>
+        </Grid>
```

Comment:
Common props should be extracted to a separate object and then spread on each component. There's too much duplication here. Duplication reduces maintainability.
E.g.
```tsx
const url = `https://modularhistory.com${collection.absoluteUrl}`;
const iconProps = { size: 36, round: true };
// ...
    <FacebookShareButton
      url={url}
      quote={collection.title}
    >
      <FacebookIcon { ...iconProps } />
    </FacebookShareButton>
// etc
```

Reply:
@iacobfred I see that the `BASE_URL` env var is set to `http://modularhistory.dev.net`, is it `https://modularhistory.com` in production? If so we can add that var to the exported `env` here
https://github.com/ModularHistory/modularhistory/blob/d04247ee7b8127390cc81629fd7a0896d37b137d/frontend/next.config.js#L76-L80
which would let us use `process.env.BASE_URL` instead of the static string.